packages/@okta/vuepress-site/docs/reference/hooks-best-practices/index.md

```diff
@@ -0,0 +1,82 @@
+
+The following sections review best practices to implement and secure Okta Event Hooks or Inline Hooks.
+
+## Secure your Hook endpoint
+
+To prevent a malicious actor from making requests to the endpoint where your Okta Hooks are sent, use the following best practices:
+
+* Configure Okta to send an authentication header in the Hook and validate it in every request by one of two ways:
+
+  * Using [HTTP Basic Authentication](/books/api-security/authn/api-authentication-options/#http-basic-authentication). When activating and enabling your Hooks on the Okta org, set the **Authorization field** as `authorization` and the **Authentication secret** in the Base64 `user:password` format.
+
+    >**Note:** You must include the authentication scheme as part of the **Authentication secret**. For Basic Authentication, your secret must appear similar to: `Basic Base64(user:password)`. See the following partial Hook header as an example:
+
+    ```cURL
+    'content-type',
+    'application/json',
+    'authorization',
+    'Basic YWRtaW46c3VwZXJzZWNyZXQ='
+    ```
+
+  * Using a separate Secret Key, which can be configured on the Hook as a custom header field.
+
+* Create an allowlist of IP addresses to check incoming Okta calls. See [What IP addresses/ranges should we allow for inbound traffic?](https://support.okta.com/help/s/article/What-IP-addresses-ranges-should-we-whitelist-for-inbound-traffic-i-e-REST-API-calls-from-Okta-to-on-prem-JIRA-server?language=en_US) for a listing of Okta IP addresses.
+
+>**Note:** This is a large list of IP addresses and the list is subject to change. Unless required by your organization, securing your Hook by authentication header is recommended.
+
+## Protect your Hook content from external viewers
+
+Okta requires HTTPS to encrypt communications to your Hook endpoint to prevent unauthorized parties from reading the contents of an Okta Hook. When using HTTPS, ensure you keep your SSL certificate updated and the Domain Name System (DNS) secured, so that someone can’t reroute your calls to another location.
+
+>**Note:** Adding an HTTP URL when enabling a Hook in the Admin Console displays an Invalid URL provided error.
+
+## Avoid delays in Hook responses
+
+When Okta uses an Inline Hook to communicate with your endpoint, the user experience is paused until your code responds. Okta Event Hooks, although acting asynchronously, also require a response. To prevent unnecessary delays or timeouts, we recommend the following:
+
+* Treat Okta Hooks as you would any other HTTP request. For example, ensure you respond to the HTTP request in under 400 milliseconds with either a 200 (Success) or 204 (Success no content) return code.
+* To avoid delays or timeouts, process the Hook request data after sending the response.
```

Comment:
This bullet point really only applies to Event Hooks since when a customer uses an Inline Hook, they need to process the Hook request data BEFORE sending back the response - I don't want to misguide customers here. Not sure if we need to make this section dedicated only to Event Hooks or clarify this bullet somehow.

Reply:
I've revised this section so the bullets will fall under Event Hooks. I recall Joel mentioned the part about Inline Hooks, so, kept it in there.